config/rbac/rbac.yaml

```diff
@@ -19,19 +19,55 @@ metadata:
   namespace: cartographer-system
 
 ---
-#!
-#!    TODO make use of granular roles rather than a catch-all "cluster-admin"
-#!         grants it more privileges than it needs.
-#!
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: cartographer-cluster-admin
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: cluster-admin
+  name: cartographer-controller-admin
 subjects:
   - kind: ServiceAccount
     name: cartographer-controller
     namespace: cartographer-system
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cartographer-controller-admin
+rules:
+  - apiGroups:
+      - carto.run
+    resources:
+      - workloads
```

Comment:
Missing deliverable and delivery, is this coming in separate PR? I'm sure we will find the answer if we keep reading

Reply:
I'd 💯 go for doing in this PR, otherwise we'd be breaking those using delivery building out of `main` (aka internal testing supplychains)

Reply:
We just removed this in the latest commit addressing the rbac roles. It seems the controller should only have permissions to update the status of workload, deliverable, delivery, etc.

If you are writing a supply chain that needs to create one of those objects, the service account you provide will need the permissions to do that.